security-csrf/src/main/java/io/micronaut/security/csrf/resolver/FutureCsrfTokenResolverAdapter.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2017-2024 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.security.csrf.resolver;
+
+import io.micronaut.core.annotation.NonNull;
+
+import java.util.concurrent.CompletableFuture;
+
+/**
+ * Adapter from {@link CsrfTokenResolver} to {@link FutureCsrfTokenResolver}.
+ * @param <T> Request
+ */
+public class FutureCsrfTokenResolverAdapter<T> implements FutureCsrfTokenResolver<T> {
```

Comment:
Is it a public and non-final class by design? No version too.

Reply:
I made it package private and final https://github.com/micronaut-projects/micronaut-security/pull/1826/commits/2fb98bfc6f69e57d712f67a92baa7bb0d1817838